### ISSUE TITLE
Charcode constants confuse hex and decimal

### ISSUE THREAD

**Issue by kleinesfilmroellchen** [python-escpos/python-escpos#358]
I have:
- [x] searched open and closed issues for duplicates: #166 #144

# Bug description

I had some problems switching to the correct code page when trying to print characters such as the euro currency sign €, latin characters with diacritics (in my case ä, ö, ü for German texts) and other characters such as 'ß'. After looking into the Constant definitions for a bit and comparing it to the code tables my printer provides (image will follow below), I found out that **the constants use the decimal number of the code page (e.g. 14), when the \x escape expects a hexadecimal number (e.g. 0e).**

### Reproduction example
The following examples assume an arbitrary escpos printer instance is assigned to the name 'printer'.

```python
# should switch to MS-DOS codepage 858 encoding (West european & euro sign)
printer.charcode('EURO')
# BUG: prints '„' (lower double 9 quote) instead of 'ä', '€' is never printed
printer.text('bär, total: 3€\n')
# set the encoding & codepage manually
printer.codepage = 'cp858'
# note the use of \x0e instead of \x14
printer._raw(b'\x1b\x74\x0e')
# this works as expected
printer.text('bär, total: 3€\n')
```

### Device info
 **Printer:** MunByn ITPP047

 **python-escpos version:** 2.2.0

 **python version:** 3.7.* (tried 3.7.3, 3.7.4)

 **operating system (uname -a):**
5.2.2-1-MANJARO #1 SMP PREEMPT Sun Jul 21 12:53:04 UTC 2019 x86_64 GNU/Linux

# Improvement Proposal

The Codepage constants are to be extended substantially to represent all available standard codepages. The Codepage constants are to be modified to set the correct codepages.

Generally, I would like to change the `charcode` function to work in two different fashions:

1. Input a number which represents the codepage to be set via the actual ESCPOS command, as well as a CodecInfo object which can convert Python strings into codepage-compatible bytestrings. Very useful if the printer has additional, weird or misplaced code pages.
2. Input a string which represents a **Python** encoding to be used. The Printer sets the codepage variable to the according `CodecInfo` by using `codecs.lookup(name)` and sends an ESCPOS command to the printer. The codepage to send to the printer will be obtained through a lookup dictionary which maps Python encodings to codepage numbers. (I am fully aware that not all Python encodings will be compatible with at least one codepage, the best example being utf-8. This is an acceptable shortcoming, as such encodings produce printing errors at the present time and would continue to do so.)

This will mean that the `codepage` variable contains a **CodecInfo** object with conversion capabilities, and not just an encoding string. This will allow the use of non-registered codecs as well as registered codecs obtainable through the `codecs.lookup(name)` method.

# Final words

As I don't currently have the time to do work on this library, I currently cannot implement the changes myself. Also, I don't know how the situation is on other printers and whether there is a general problem with the constants (or is it just me? maybe). When I find some time, I will definitely implement the above changes and attach PRs to this issue. When some does it before me: Great, thanks!

I am very open to suggestions and/or changes regarding the API change proposed above.

Thank you all very much in advance, kleinesfilmröllchen

PS: image of codetables:

![grafik](https://user-images.githubusercontent.com/28656157/62950235-f2327980-bde7-11e9-8fc1-7376ca89775d.png)

PPS: if you were guessing, yes, I am German, please don't judge my English too hard :)

**Comment by kleinesfilmroellchen** [comment]
It occurs to me that most of the time, instead of the unknown characters, nothing is printed at all, so expect that result as well if you are testing this.

**Comment by kleinesfilmroellchen** [comment]
Proposed Labels: bug easy

**Comment by patkan** [comment]
Hi, thank you for the extensive report. Have you tried more current versions of the project than v2.2?

**Comment by kleinesfilmroellchen** [comment]
Hey, sorry for leaving this dormant for a year.

I have not worked with the printer over the last year because the project had been finalized and we'll only need it to run it in about half a year. Also, I don't have it on hand right now, so I can't check whether upgrading to escpos 3+ will work. I'll however put a (mental) pin in this and do the upgrade tests as soon as I get the printer back.

**Comment by dalpil** [comment]
I ran into this issue while trying to print Swedish text (with "ÅÄÖ/åäö" characters) on the Munbyn ITPP047 with python-escpos 2.2.0.

The issue seems to be fixed in 3.0a8 though, as I don't have to specify a codepage manually for my characters to be printed correctly.

**Comment by patkan** [comment]
No further feedback, seems to be fixed in v3
